vllm/worker/model_runner_base.py

```diff
@@ -36,15 +39,19 @@ def _init_attn_metadata_from_tensor_dict(
     Helper method to initialize AttentionMetadata based on an
     AttentionBackend and broadcastable AttentionMetadata fields.
     """
-    # Extract the fields used to create AttentionMetadata.
-    valid_attn_kwargs = {}
-    for field in dataclasses.fields(attn_backend.get_metadata_cls()):
-        val = tensor_dict.pop(field.name, None)
-        if val is not None:
-            valid_attn_kwargs[field.name] = val
-
-    attn_metadata = attn_backend.make_metadata(**valid_attn_kwargs)
-    tensor_dict["attn_metadata"] = attn_metadata
+    skip_attn_metadata = tensor_dict.pop("skip_attn_metadata", False)
+    if skip_attn_metadata:
+        tensor_dict["attn_metadata"] = None
```

Comment:
without the explicit assigning, the `tensor_dict` doesn't have `attn_metadata` key and leads to error, right?

Reply:
When we get a chance, we probably need to check more field similar like this. My concern is skipping logic may introduce more inconsistency than we know

Reply:
> without the explicit assigning, the `tensor_dict` doesn't have `attn_metadata` key and leads to error, right?

more precisely is `tensor_dict` doesn't have all the parameters to build `attn_metadata`, which corresponds to this issue we observed before 
![image](https://github.com/user-attachments/assets/7f6d1f75-63f2-4dd0-b396-f59da7251a5f)


> When we get a chance, we probably need to check more field similar like this. My concern is skipping logic may introduce more inconsistency than we know

yes, I also have the same concern. I don't know whether vLLM has unit tests to cover all the features. If so, we could leverage those UTs (by enabling kv cache) to check if the skipping logic breaks any of them

Reply:
It does have enough coverage but we might need to add more code changes to adopt the cache solution first. Let's run one more round stability test due to limited time and check those issues from high level. If we still encounter lots of issue, then we have to dig into the unit test level.

Reply:
Since we've passed several rounds of stability tests, I'm going to merge the PR and will follow up with adding more UTs to cover features related to kv cache later.